### PR TITLE
gadgets: Add top_tcp as image based gadget

### PR DESCRIPTION
**Pull request by burak-ok**
# Add top_tcp as image based gadget

# Also tested IPv6 and filtering:
```
RUNTIME.CONTAINERNAME           PID          TID SRC                            DST                            COMM        SENT        RECEIVED   
                             223189       223189 ::1:8080                       ::1:55842                      nc          0           74         
                             223202       223202 ::1:55842                      ::1:8080                       curl        74          0
```

### CODE REVIEW

File: gadgets/top_tcp/test/top_tcp_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type topTcpEntry struct {
+	eventtypes.CommonData
+
+	MountNsID uint64 `json:"mntns_id"`
+
+	Pid  uint32 `json:"pid"`
+	Tid  uint32 `json:"tid"`
```
Comment (blanquicet) [inline gadgets/top_tcp/test/top_tcp_test.go:38]: Could you please add also [uid/gid](https://github.com/inspektor-gadget/inspektor-gadget/pull/3207#discussion_r1693302671)?

File: gadgets/top_tcp/test/top_tcp_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+func TestTopTcp(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-top-tcp"
+	containerImage := "docker.io/library/busybox:latest"
```
Comment (blanquicet) [inline gadgets/top_tcp/test/top_tcp_test.go:57]: This should be `docker.io/library/nginx:latest`.

File: gadgets/top_tcp/test/top_tcp_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+						Port:    utils.NormalizedInt,
+						Proto:   6,
+					},
+					Dst: utils.L4Endpoint{
+						Addr:    "127.0.0.1",
+						Version: 4,
+						Port:    utils.NormalizedInt,
+						Proto:   6,
+					},
+					Comm: "sh",
```
Comment (blanquicet) [inline gadgets/top_tcp/test/top_tcp_test.go:122]: Mmmm why `sh`? It should be `curl` with dst.port 80 or `nginx` with src.port 80. Or, you can normalize it to match both.

**Review by blanquicet — COMMENTED** [review]
The gadget looks good, but the test still need some tune.

File: gadgets/top_tcp/program.bpf.c [hunk 4]
```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Francis Laniel <flaniel@linux.microsoft.com>
```
[the remark this answers is not in the document]
Comment (burak-ok) [reply gadgets/top_tcp/program.bpf.c:2]: I left it like it was in the built-in gadget. Also I don't know if I am "allowed" to change that :sweat_smile:

File: gadgets/top_tcp/test/top_tcp_test.go (hunk 3 above)
Comment (burak-ok) [reply gadgets/top_tcp/test/top_tcp_test.go:122]: Also copied it form the built-in gadget test

File: gadgets/top_tcp/program.bpf.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (blanquicet) [reply gadgets/top_tcp/program.bpf.c:2]: cc @mauriciovasquezbernal

File: gadgets/top_tcp/test/top_tcp_test.go (hunk 1 above)
Comment (blanquicet) [reply gadgets/top_tcp/test/top_tcp_test.go:38]: This can be done in a follow-up PR if you prefer.

**Review by blanquicet — APPROVED** [review]
LGTM! I tested on top of combiner and it worked correctly.

Columns version not working because of issue with `gadget_l4endpoint_t`.

**Pull request merged**